yt_dlp/extractor/niconico.py

```diff
@@ -898,11 +894,16 @@ def _real_extract(self, url):
                     'abr': abr,
                     'format_id': a_map.get(abr, fmt['format_id']),
                 })
-            fmt.update(fmt_common)
 
         return {
             'id': video_id,
             'title': title,
+            'downloader_options': {
+                'max_quality': traverse_obj(embedded_data, (
+                    'program', 'stream', 'maxQuality', {str})),
+                'ws': ws,
+                'ws_url': ws_url,
+            },
```

Comment:
I kinda like moving this to `downloader_options` (and I assume it was necessary for tests to pass); however, we should document these in the `InfoExtractor` docstring, and we should set a default value for `max_quality` (`'high'`?)
```suggestion
            'downloader_options': {
                'max_quality': traverse_obj(embedded_data, ('program', 'stream', 'maxQuality', {str})) or 'high',
                'ws': ws,
                'ws_url': ws_url,
            },
```

Reply:
```diff
diff --git a/yt_dlp/extractor/common.py b/yt_dlp/extractor/common.py
index 6058f66ae..32b4680b7 100644
--- a/yt_dlp/extractor/common.py
+++ b/yt_dlp/extractor/common.py
@@ -263,6 +263,9 @@ class InfoExtractor:
                                  * http_chunk_size Chunk size for HTTP downloads
                                  * ffmpeg_args     Extra arguments for ffmpeg downloader (input)
                                  * ffmpeg_args_out Extra arguments for ffmpeg downloader (output)
+                                 * ws              (NiconicoLiveFD only) WebSocketResponse
+                                 * ws_url          (NiconicoLiveFD only) Websockets URL
+                                 * max_quality     (NiconicoLiveFD only) Max stream quality string
                     * is_dash_periods  Whether the format is a result of merging
                                  multiple DASH periods.
                     RTMP formats can also have the additional fields: page_url,
```